addons/account/models/account_move.py

```diff
@@ -90,7 +90,6 @@ def _sequence_fixed_regex(self):
         string='Date',
         index=True,
         compute='_compute_date', store=True, required=True, readonly=False, precompute=True,
-        states={'posted': [('readonly', True)], 'cancel': [('readonly', True)]},
```

Comment:
It is so much more interesting to have `readonly` declared in the module by default (overridable in the view) in order to have the ORM constraint writes on readonly fields.
Even if we don't consider the constraint for various reasons, it makes more sense to declare all the readonly stuff at one place only (instead of `readony=False` here and then on the view)

Reply:
There was never any constraint using states in the ORM, only for the web client. The places where there was code to check states was wrong, it didn't even check the second argument.

Readonly=False is used to signal that a compute can be used as an onchange and the field accepts writing.

It is precisely to remove all this confusion and misuse that this change is made. The idea of putting a domain instead of readonly has been discussed in framework. The pr was ready and analyzed, after which this solution was rejected.
cc @rco-odoo & @VincentSchippefilt

Reply:
During my R&D for this task, I had looked for the view and the python model to be merged with a '`or`' (so `readonly=True` on the model would have remained True). The choice (al, fp) is to not change that and make the view always overwrite the python value.